zerver/tests/test_messages.py

```diff
@@ -3164,6 +3166,141 @@ def test_propagate_all_topics(self) -> None:
+                                          topic_name="test", content="First")
+        self.send_stream_message(user_profile.email, stream.name,
+                                 topic_name="test", content="Second")
+
+        result = self.client_patch("/json/messages/" + str(msg_id), {
+            'message_id': msg_id,
+            'new_stream_id': new_stream.id,
+            'propagate_mode': 'change_all',
+            'content': 'Not allowed'
+        })
+
+        messages = get_topic_messages(user_profile, stream, "test")
+        self.assertEqual(len(messages), 2)
+
+        messages = get_topic_messages(user_profile, new_stream, "test")
+        self.assertEqual(len(messages), 0)
+
+        self.assert_json_error(result, "Cannot change message content while changing stream")
+
+    def test_move_message_to_stream_and_topic_fail(self) -> None:
+        user_profile = self.example_user("iago")
+        self.login(user_profile.email)
+        stream = self.make_stream("test move stream")
+        new_stream = self.make_stream("new stream")
+        self.subscribe(user_profile, stream.name)
+        self.subscribe(user_profile, new_stream.name)
+        msg_id = self.send_stream_message(user_profile.email, stream.name,
+                                          topic_name="test", content="First")
+        self.send_stream_message(user_profile.email, stream.name,
+                                 topic_name="test", content="Second")
+
+        result = self.client_patch("/json/messages/" + str(msg_id), {
+            'message_id': msg_id,
+            'new_stream_id': new_stream.id,
+            'propagate_mode': 'change_all',
+            'topic': 'Not allowed'
+        })
+
+        messages = get_topic_messages(user_profile, stream, "test")
+        self.assertEqual(len(messages), 2)
+
+        messages = get_topic_messages(user_profile, new_stream, "test")
+        self.assertEqual(len(messages), 0)
+
+        self.assert_json_error(result, "Cannot change topic while changing stream")
+
+    def test_move_message_to_stream_to_private_stream(self) -> None:
+        user_profile = self.example_user("iago")
+        self.login(user_profile.email)
+        stream = self.make_stream("test move stream")
+        new_stream = self.make_stream("new stream", None, True)
+        self.subscribe(user_profile, stream.name)
+        self.subscribe(user_profile, new_stream.name)
+        msg_id = self.send_stream_message(user_profile.email, stream.name,
+                                          topic_name="test", content="First")
+        self.send_stream_message(user_profile.email, stream.name,
+                                 topic_name="test", content="Second")
+
+        result = self.client_patch("/json/messages/" + str(msg_id), {
+            'message_id': msg_id,
+            'new_stream_id': new_stream.id,
+            'propagate_mode': 'change_all',
+        })
+
+        messages = get_topic_messages(user_profile, stream, "test")
```

Comment:
Add a comment above this noting that we expect the messages to remain in the original stream/topic.